src/vm/gcenv.os.cpp

```diff
@@ -498,7 +506,40 @@ static size_t GetRestrictedPhysicalMemoryLimit()
     {
         job_physical_memory_limit = 0;
 
-        FreeLibrary(hinstKernel32);
+        if (hinstKernel32 != 0)
+        {
+            FreeLibrary(hinstKernel32);
+            hinstKernel32 = 0;
+            GCGetProcessMemoryInfo = 0;
+        }
+    }
+
+    // Check to see if we are limited by VM.
+    if (total_virtual == 0)
+    {
+        MEMORYSTATUSEX ms;
+        ::GetProcessMemoryLoad(&ms);
+
+        total_virtual = ms.ullTotalVirtual;
+        total_physical = ms.ullTotalPhys;
+    }
+
+    if (job_physical_memory_limit != 0)
+    {
+        total_physical = job_physical_memory_limit;
+    }
+
+    if (total_virtual < total_physical)
+    {
+        if (hinstKernel32 != 0)
+        {
+            // We can also free the lib here - if we are limited by VM we will not be calling
```

Comment:
Do we really need to load K32GetProcessMemoryInfo dynamically? According to MSDN, it is always available in kernel32.dll starting with Windows 7. The dynamic loading looks like leftover from times when we have supported earlier OSes.

And CoreCLR.dll always depends on kernel32.dll, so we are not getting anything by loading it dynamically.

Reply:
I just looked; this code was changed a lot by various people and they gradually got rid of the dynamic checks...but somehow [the last checkin](https://github.com/dotnet/coreclr/pull/8624) left this one...which I didn't CR 'cause I was out (but you did :-))...

Reply:
Similar cleanups tend to be done gradually. People do an iterations and then move on, even though there is more that can be cleaned up. 

I have just pointed this out since you are adding more code here that does not seem necessary.

Reply:
it's only unnecessary if the dynamic check isn't there. do you remember if there was a reason why this particular dynamic check was left there? it would seem it's easier to just remove them all.

Reply:
https://github.com/dotnet/coreclr/pull/8624 was about removing API sets dependencies.

The dynamic check for K32GetProcessMemoryInfo was there even on the non-API set (ie `!FEATURE_CORECLR`) codepath before the change, and I guess that's why it was kept even with API set dependencies removed.

Reply:
ok...I'm going to open an issue to track this - I also see that there're unnecessary things like "GCIsProcessInJob = &(::IsProcessInJob);", why not just call IsProcessJob if we are sure it's there? also the standalone gc's implementation still has all the checks which seem unnecessary.